lib/builtins/deploy-delegates/lambda-deployer/helper.js

```diff
@@ -10,56 +10,86 @@ const retryUtils = require('@src/utils/retry-utility');
 const stringUtils = require('@src/utils/string-utils');
 
 module.exports = {
-    validateLambdaDeployState,
+    loadLambdaInformation,
     deployIAMRole,
     deployLambdaFunction
 };
 
-function validateLambdaDeployState(reporter, awsProfile, awsRegion, currentRegionDeployState, ignoreHash, callback) {
-    const lambdaData = currentRegionDeployState.lambda;
-    if (R.isNil(lambdaData) || R.isEmpty(lambdaData) || R.isNil(lambdaData.arn)) {
-        return callback(null, { updatedDeployState: currentRegionDeployState });
+function loadLambdaInformation(reporter, loadLambdaConfig, callback) {
+    const { awsProfile, awsRegion, alexaRegion, deployState, userConfig, ignoreHash } = loadLambdaConfig;
+    const lambdaClient = new LambdaClient({ awsProfile, awsRegion });
+    // check if user sets userConfig with "sourceLambda" field to reuse existing Lambda
+    const sourceLambdaArn = alexaRegion === 'default' ? R.view(R.lensPath(['sourceLambda', 'arn']), userConfig)
+        : R.view(R.lensPath(['regionalOverrides', alexaRegion, 'sourceLambda', 'arn']), userConfig);
+    if (stringUtils.isNonBlankString(sourceLambdaArn)) {
+        return _loadLambdaFromUserConfigSource(reporter, lambdaClient, sourceLambdaArn, (loadErr, sourceLambdaData) => {
+            callback(loadErr, loadErr ? undefined : sourceLambdaData);
+        });
+    }
+
+    // Lambda does not exist in deployState
+    const lambdaArn = R.view(R.lensPath(['lambda', 'arn']), deployState);
+    if (R.isNil(lambdaArn)) {
+        return callback(null, {});
     }
+
+    // Lambda exists, validate the deploy states
     reporter.updateStatus('Validating the deploy state of existing Lambda function...');
-    let updatedDeployState;
-    const lambdaClient = new LambdaClient({ awsProfile, awsRegion });
-    const lambdaArn = lambdaData.arn;
     lambdaClient.getFunction(lambdaArn, (err, data) => {
         if (err) {
             return callback(err);
         }
         // 1. check IAM role arn
-        const localIAMRole = currentRegionDeployState.iamRole;
+        const localIAMRole = deployState.iamRole;
         const remoteIAMRole = data.Configuration.Role;
         if (stringUtils.isNonBlankString(localIAMRole) && !R.equals(localIAMRole, remoteIAMRole)) {
-            return callback(`The IAM role for Lambda ARN (${lambdaArn}) should be ${remoteIAMRole}, but found ${localIAMRole}. \
+            return callback(`The IAM role for Lambda ARN (${lambdaArn}) should be "${remoteIAMRole}", but found "${localIAMRole}". \
```

Comment:
optional(can be fixed during rewrite of lambda deployer)
should be `new callback(new CliError(message))`?

same in other places where error is thrown.

Reply:
For this error class usage, fixing in the rewrite would be a better idea to me. I have added a TODO.